swift/Samples/SplitScreenContainer/Sources/SplitScreenContainerViewController.swift

```diff
@@ -0,0 +1,127 @@
+        view.addSubview(leftContainerView)
+        
+        separatorView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(separatorView)
+        
+        rightContainerView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(rightContainerView)
+
+        NSLayoutConstraint.activate([
+            leftContainerView.topAnchor.constraint(equalTo: view.topAnchor),
+            leftContainerView.leftAnchor.constraint(equalTo: view.leftAnchor),
+            leftContainerView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            leftContainerView.rightAnchor.constraint(equalTo: separatorView.leftAnchor),
+
+            separatorView.topAnchor.constraint(equalTo: view.topAnchor),
+            separatorView.widthAnchor.constraint(equalToConstant: 1.0),
+            separatorView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+
+            rightContainerView.topAnchor.constraint(equalTo: view.topAnchor),
+            rightContainerView.leftAnchor.constraint(equalTo: separatorView.rightAnchor),
+            rightContainerView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            rightContainerView.rightAnchor.constraint(equalTo: view.rightAnchor),
+        ])
+        
+        updateWidthConstraints(ratio: screen.ratio, animated: false)
+        
+        self.leftContentViewController = embed(screen.leftScreen, in: leftContainerView)
+        self.rightContentViewController = embed(screen.rightScreen, in: rightContainerView)
+        
+        update(with: screen)
+    }
+    
+    private func embed<ScreenType: Screen>(_ screen: ScreenType, in containerView: UIView) -> ScreenViewController<ScreenType> {
+        let viewController = viewRegistry.provideView(for: screen)
+        
+        addChild(viewController)
+        
+        viewController.view.translatesAutoresizingMaskIntoConstraints = false
+        containerView.addSubview(viewController.view)
+        
+        NSLayoutConstraint.activate([
+            viewController.view.topAnchor.constraint(equalTo: containerView.topAnchor),
+            viewController.view.rightAnchor.constraint(equalTo: containerView.rightAnchor),
+            viewController.view.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
+            viewController.view.leftAnchor.constraint(equalTo: containerView.leftAnchor),
+            ])
+        
+        viewController.didMove(toParent: self)
+        
+        return viewController
+    }
+    
+    private func updateWidthConstraints(ratio: CGFloat, animated: Bool) {
+        func updateConstraints() {
+            leftContainerViewWidthConstraint?.isActive = false
+            
+            leftContainerViewWidthConstraint = leftContainerView.widthAnchor.constraint(equalTo: view.widthAnchor, multiplier: ratio)
+            leftContainerViewWidthConstraint?.isActive = true
+        }
+        
+        guard animated else {
```

Comment:
Not sure I would guard here. If someone comes in later to add more constraints logic, it might not be obvious that the guard is only used to do animation.

Reply:
Resolved via https://github.com/square/workflow/pull/925/commits/809bb987cf6e6d05e99024f21a7dc28fb86a70f8